mapiproxy/servers/default/emsmdb/oxcfxics.c

```diff
@@ -3364,13 +3324,11 @@ _PUBLIC_ enum MAPISTATUS EcDoRpc_RopSyncImportReadStateChanges(TALLOC_CTX *mem_c
 	struct emsmdbp_object			*synccontext_object, *folder_object, *message_object;
 	enum MAPISTATUS				retval;
 	enum mapistore_error			ret;
-	struct MessageReadState			*read_states;
-	uint32_t				read_states_size;
-	struct Binary_r				*bin_data;
+	struct MessageReadState			*msg_read_state;
 	char					*owner;
 	uint64_t				mid, base;
 	uint16_t				replid;
-	int					i;
+	int					i, j;
```

Comment:
Because there is a nested `for` at L3405. First, `i` is for going through `MessageReadStates` and the second one is for getting the `mid` from the source key.